### PR TITLE
New resource to manage a project's billing account

### PR DESCRIPTION
**Pull request by juliocc**
Add new resource and data source to manage a project's billing account.

Fixes hashicorp/terraform-provider-google#3792

If this PR is for Terraform, I acknowledge that I have:

- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [X] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.
- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
`google_billing_project_info`
```

### CODE REVIEW

**Comment by juliocc** [comment]
This is ready for review.

I tried creating a data source too but I the generated resource is in a different package and can't reuse the schema and read operations. If you know an easy way around that, I can try it again.

**Comment by juliocc** [comment]
@slevenick can you confirm if the test failures are related to my changes?

Thanks

File: mmv1/products/billing/ProjectInfo.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Resource
+name: ProjectInfo
+base_url: 'projects/{{project}}/billingInfo'
+create_verb: :PUT
```
Comment (slevenick) [inline mmv1/products/billing/ProjectInfo.yaml:17]: Is this resource updatable as well? If so, can you add a handwritten update test? If not, you can add immutable: true here

**Review by slevenick — CHANGES_REQUESTED** [review]

File: mmv1/products/billing/ProjectInfo.yaml (hunk 1 above)
Comment (juliocc) [reply mmv1/products/billing/ProjectInfo.yaml:17]: It is updatable. 

I'll add the update test and update the PR

**Comment by slevenick** [comment]
> @slevenick can you confirm if the test failures are related to my changes?
> 
> Thanks

The test failures are unrelated.

File: mmv1/products/billing/ProjectInfo.yaml (hunk 1 above)
Comment (juliocc) [reply mmv1/products/billing/ProjectInfo.yaml:17]: Done

**Review by slevenick — APPROVED** [review]
Thanks! Looks like tests pass in CI

**Pull request merged**